crates/astria-composer/src/executor/mod.rs

```diff
@@ -415,6 +419,23 @@ impl Executor {
 
         reason.map(|_| ())
     }
+
+    // check for mismatched configured chain_id and sequencer client chain_id
+    pub(crate) async fn check_chain_ids(&self) -> eyre::Result<()> {
+        let client_response = self
+            .sequencer_client
+            .status()
+            .await
+            .wrap_err("failed to retrieve sequencer network status")?;
+        let client_chain_id = client_response.node_info.network.to_string();
+        let configured_chain_id = self.sequencer_chain_id.clone();
```

Comment:
You can avoid the clones/allocations. The `Genesis` return value has a `chain_id` field that provides a `as_str()` method:
https://docs.rs/tendermint/0.37.0/tendermint/genesis/struct.Genesis.html
https://docs.rs/tendermint/0.37.0/tendermint/chain/id/struct.Id.html#method.as_str

The check below would then become:
```rust
ensure!(
    &self.sequencer_chain_id == genesis.chain_id.as_str(), "<text>"
);
```

Reply:
Resolved in f3c96ff.